### PR TITLE
[bitnami/external-dns] Add aws.zoneMatchParent option

### PR DESCRIPTION
**Pull request by clemensg**
### Description of the change

Add support for the aws.zoneMatchParent option (defaulting to false) and add corresponding documentation.

### Benefits

Users can set aws.zoneMatchParent to true and set a domain filter for subdomains, e.g. a.b.example.com. This was not possible before with external-dns and Route53.
See https://github.com/kubernetes-sigs/external-dns/pull/4236

### Possible drawbacks

None

### Applicable issues

None

### Additional information

N/A

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ maintainers:
 name: external-dns
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/external-dns
-version: 7.2.1
+version: 7.2.2
```
Comment (dgomezleon) [inline bitnami/external-dns/Chart.yaml:31]: Please bump the minor version to add new values.
```suggestion
version: 7.3.0
```

**Review by dgomezleon — COMMENTED** [review]
Thanks for contributing. Just a minor change regarding the version.

File: bitnami/external-dns/Chart.yaml (hunk 1 above)
Comment (clemensg) [reply bitnami/external-dns/Chart.yaml:31]: Will do

**Review by dgomezleon — DISMISSED** [review]
LGTM

**Review by dgomezleon — APPROVED** [review]

**Pull request merged**